### PR TITLE
Custom fallback

### PR DESCRIPTION
**Pull request by teseoch**

### CODE REVIEW

**Review by zfergus — APPROVED** [review]

File: src/polysolve/nonlinear/Solver.cpp [hunk 1]
```diff
@@ -24,6 +24,75 @@
 
 namespace polysolve::nonlinear
 {
+    namespace
+    {
+        std::shared_ptr<DescentStrategy> crate_solver(
```
Comment (Huangzizhou) [inline src/polysolve/nonlinear/Solver.cpp:29]: typo

**Review by Huangzizhou — APPROVED** [review]

File: nonlinear-solver-spec.json [hunk 2]
```diff
@@ -263,6 +263,241 @@
...
+        ],
+        "optional": [
+            "residual_tolerance"
+        ],
+        "doc": "Options for Newton."
+    },
+    {
+        "pointer": "/solver/*",
+        "type": "object",
+        "type_name": "RegularizedNewton",
```
[the remark this answers is not in the document]
Comment (teseoch) [reply nonlinear-solver-spec.json:306]: done, I also added the new type

File: nonlinear-solver-spec.json [hunk 3]
```diff
@@ -263,6 +263,241 @@
         "type": "float",
         "doc": "Probability of erasing a component on the gradient for StochasticGradientDescent."
     },
+    {
+        "pointer": "/solver",
```
[the remark this answers is not in the document]
Comment (teseoch) [reply nonlinear-solver-spec.json:275]: it is on top level with the rest, why should we move it?

File: nonlinear-solver-spec.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (Huangzizhou) [reply nonlinear-solver-spec.json:306]: thanks!

File: nonlinear-solver-spec.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (Huangzizhou) [reply nonlinear-solver-spec.json:275]: Oh I see, it's used in Solver, nvm

**Pull request merged**